src/proxyvisor.coffee

```diff
@@ -0,0 +1,208 @@
+				id: device.deviceId
+				body: { status, is_online }
+				customOptions:
+					apikey: apiKey
+			.then ->
+				device.status = status
+				device.is_online = is_online
+				knex('dependentDevice').update(device).where({ uuid })
+				.then ->
+					res.json(device)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+tarPath = (app) ->
+	return '/tmp/' + app.commit + '.tar'
+
+router.get '/v1/assets/:commit', (req, res) ->
+	knex('dependentApp').select().where({ commit: req.params.commit })
+	.then ([ app ]) ->
+		return res.status(404).send('Not found') if !app
+		dest = tarPath(app)
+		getAssetsPath(app.imageId)
+		.then (path) ->
+			getTarArchive(path, dest)
+		.then ->
+			res.sendFile(dest)
+	.catch (err) ->
+		console.error(err)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+getTarArchive = (path, destination) ->
+	fs.lstatAsync(path)
+	.then ->
+		execAsync("cd #{path} && tar -cvf #{destination} *")
+
+# TODO: deduplicate code from compareForUpdate in application.coffee
+exports.fetchAndSetTargetsForDependentApps = (state, fetchFn) ->
+	knex('dependentApp').select()
+	.then (localDependentApps) ->
+		# Compare to see which to fetch, and which to delete
+		remoteApps = _.mapValues state.apps, (app, appId) ->
+			env = app.environment ? {}
+			return {
+				appId: appId
+				imageId: app.image
+				commit: app.commit
+				env: JSON.stringify(env)
+			}
+		localApps = _.indexBy(localDependentApps, 'appId')
+
+		toBeDownloaded = _.values(
+			_.filter remoteApps, (app, appId) ->
+				return  app.commit? and !_.any(localApps, imageId: app.imageId)
+		)
+		toBeRemoved = _.values(
+			_.filter localApps, (app, appId) ->
+				return app.commit? and !_.any(remoteApps, imageId: app.imageId)
+		)
+		Promise.map toBeDownloaded, (app) ->
+			fetchFn(app, false)
+		.then ->
+			Promise.map toBeRemoved, (app) ->
+				fs.unlinkAsync(tarPath(app))
+				.then ->
+					docker.getImage(app.imageId).removeAsync()
+				.catch (err) ->
+					console.error('Could not remove image/artifacts for dependent app', err, err.stack)
+		.then ->
+			Promise.props(
+				_.mapValues remoteApps, (app, appId) ->
+					knex('dependentApp').update(app).where({ appId })
+					.then (n) ->
+						knex('dependentApp').insert(app) if n == 0
+			)
+		.then ->
+			devices = _.map state.devices, (device, uuid) ->
+				device.uuid = uuid
+				return device
+			Promise.map devices, (device) ->
+				# Only consider one app per dependent device for now
+				appId = _.keys(device.apps)[0]
+				knex('dependentDevice').update({ targetEnv: JSON.stringify(device.environment), targetCommit: state.apps[appId].commit }).where({ uuid: device.uuid })
+	.catch (err) ->
+		console.error('Error fetching dependent apps', err, err.stack)
+
+sendUpdate = (device) ->
+	request.putAsync("#{config.proxyvisorHookReceiver}/v1/devices/#{device.uuid}", { json: true, body: { commit: device.targetCommit, environment: JSON.parse(device.targetEnv) } })
```

Comment:
I think it would make sense to break this object onto multiple lines for readability